### PR TITLE
fix(build): ensure the dist files' directories exist before writing them

### PR DESCRIPTION
**Pull request by symful**
If files from static folder are in subdirectories, the CLI will error'd. This ensure that the directories of dist files (including from static folder), exist before writing the files.

### CODE REVIEW

**Comment by kt3k** [comment]
(CI issue is irrelevant of this change. I'll fix them and merge this soon)

**Comment by NekoMaru76** [comment]
Fixed the test.
Also I added flag `config` (`c`), it is the path for `tsconfig.json`. I added this to enable to customize the `jsx`, `jsxFragment`, `jsxFactory`, etc. The getter function (located in `bundle_util.ts` will first look for `deno.json`, `denoc.json`, or `tsconfig.json` if the flag is not set. I made it so it manually read the config file and manually put the `jsx` options from the config file since it doesn't work if I just put the `tsconfig` path. I don't know why :/

File: bundle_util.ts [hunk 1]
```diff
@@ -36,3 +61,33 @@ export function setImportMap(importMap: string) {
...
+
+export async function getTsconfig() {
+  if (!_tsconfig) {
+    switch (true) {
+      case await exists("./deno.json", {
+        isReadable: true,
+        isDirectory: false,
+      }):
+        return "./deno.json";
+      case await exists("./denoc.json", {
```
Comment (kt3k) [inline bundle_util.ts:79]: This should be `deno.jsonc` and it can be have comments. (`JSON.parse` will cause parse error in that case)

File: bundle_util.ts [hunk 2]
```diff
@@ -36,3 +61,33 @@ export function setImportMap(importMap: string) {
...
+
+let _tsconfig: string | undefined;
+
+export function setTsconfig(tsconfig: string) {
+  _tsconfig = tsconfig;
+}
+
+export async function getTsconfig() {
+  if (!_tsconfig) {
+    switch (true) {
```
Comment (kt3k) [inline bundle_util.ts:73]: This idiom looks confusing. Can you rewrite this to `if-else` blocks

File: generate_static_assets.ts [hunk 3]
```diff
@@ -77,8 +77,10 @@ async function createStaticAssetFromPath(
 ): Promise<File> {
   logger.debug("Reading", path);
   const bytes = await Deno.readFile(path);
+  const webkitRelativePath = relative(root, path);
   return Object.assign(new Blob([bytes]), {
-    name: join(distPrefix, relative(root, path)),
+    name: join(distPrefix, webkitRelativePath),
     lastModified: 0,
+    webkitRelativePath,
```
Comment (kt3k) [inline generate_static_assets.ts:84]: Why is this necessary?

Comment (NekoMaru76) [reply generate_static_assets.ts:84]: image
It said `File` needs to has [webkitRelativePath](https://developer.mozilla.org/en-US/docs/Web/API/File/webkitRelativePath).

File: bundle_util.ts (hunk 1 above)
Comment (NekoMaru76) [reply bundle_util.ts:79]: Then I will add another module to `deps.ts` ([jsonc](https://deno.land/std@0.193.0/jsonc/mod.ts))

File: bundle_util.ts (hunk 2 above)
Comment (NekoMaru76) [reply bundle_util.ts]: Done!

File: generate_static_assets.ts (hunk 3 above)
Comment (NekoMaru76) [reply generate_static_assets.ts:84]: Though, do you think that it is better to just use `File` constructor? Since you want it to return `File` object so, will you let me refactor all of the `Object.assign` + `new Blob`?

Comment (kt3k) [reply generate_static_assets.ts:84]: I think it's fine for now. Thanks for updating!

Comment (NekoMaru76) [reply generate_static_assets.ts:84]: Though I already did it in my local, mind if I make another pull request regarding this?

Comment (kt3k) [reply generate_static_assets.ts:84]: BTW while I'm fine with `webkitRelativePath` (it's at least harmless), I'm curious where that type error comes from because Deno and TypeScript don't seem having that property defined in File. https://github.com/search?q=repo%3Adenoland%2Fdeno%20webkitRelativePath&type=code

Comment (kt3k) [reply generate_static_assets.ts:84]: > Though I already did it in my local, mind if I make another pull request regarding this?

Yes, that's also welcome!

Comment (NekoMaru76) [reply generate_static_assets.ts:84]: > 

[lib.dom.ts](https://github.com/denoland/deno/blob/0c3bbf7acdeacdd1c36b4b847679d85ca19962dc/cli/tsc/dts/lib.dom.d.ts#L8220)
